source/fab/steps/compile_c.py

```diff
@@ -114,7 +114,7 @@ def _compile_file(self, analysed_file: AnalysedFile):
                 try:
                     run_command(command)
                 except Exception as err:
-                    return TaskException(f"error compiling {analysed_file.fpath}: {err}")
+                    return TaskException(f"error compiling {analysed_file.fpath}:\n{err}")
```

Comment:
Might it be worth creating our own exception class which takes a message, file and exception on its constructor? Then its `__str__` method could do this formatting across the board.

Reply:
Yes, I think there's likely a few useful Fab exceptions we could make. There's a ticket to review the exception handling. I'd suggest that would fall under it's remit?